README.md

```diff
@@ -117,47 +120,171 @@ If you'd like to get a message when the button is pressed or released, call the
 
     # Make sure that you've enabled or loaded the SPI driver or this will
     # fail.
     iex> alias ElixirALE.SPI
     iex> {:ok, pid} = SPI.start_link("spidev0.0")
     {:ok, #PID<0.124.0>}
 
-    # Read the potentiometer
+    # Read the moisture sensor
+
+    # Use binary pattern matching to pull out the ADC counts (low 10 bits)
+    iex> mode = 1 #Single-ended mode
+    iex> sensor = 2 #Try the CH2 sensor
+    iex> <<_::size(14), value::size(10)>> =
+           SPI.transfer(pid, <<0x01, mode::size(1), sensor::size(3), 0::size(12)>>)
+    TODO "NEED WHAT IT RETURNS"
+
+    iex> IO.puts "Sensor #{sensor} measured #{value}"
+    TODO "NEED WHAT IT RETURNS"
+
+If you are new to using Elixir for embedded, there may be some things here you 
+might not be familiar with.  We'll be working with bit strings.  If this is 
+new to you, you might want to refer to the 
+[Elixir docs for bitstings](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#%3C%3C%3E%3E/1).
+
+First we call SPI.start_link with Linux spidev device driver as it argument.  
+We pattern match with `{:ok, pid}` so we can use `pid` in the 
+SPI.transfer call next.
+
+    {:ok, pid} = SPI.start_link("spidev0.0")
+
+Elixir ALE uses the Linux spidev device driver for communicating with SPI 
+devices. It creates files in /dev for each SPI interface that it knows about. 
+Many boards only have one SPI interface and it shows up as /dev/spidev0.0. 
+The easiest way to find out the right name is to run ls spidev* in the /dev 
+directory. If nothing shows up, it is likely that the spidev driver needs to 
+be loaded. Try running modprobe spidev or consult your board's documentation.
+
+Next we'll make a call to SPI.transfer to read the moisture sensor.
+
+    SPI.transfer(pid, <<0x01, mode::size(1), sensor::size(3), 0::size(12)>>)
+
+We are building the bitstring in 4 segments.  The first segement `0x01`, is the 
+hexadecimal representation of `00000001` that we determined above we need to 
+send in the first byte.  The final bit `1` will be the start bit.  It is 
+common to describe bytes using hexadecimal notation.
+
+The next bit will be for the mode which we have set to 1 for Single-ended mode.
+
+TODO:  The slides say 0 for Single-ended mode, but that doesnt' seem to match 
+the data sheet. Need to verify which is correct.
+
+Next we will send 3 bits to select the sensor on channel 2.  As we found above in 
+Table 5-2 on pg 19 of the data sheet `010` will select channel 2 
+(CH2).  `2::size(3)` is equivalent to `010` and sends the binary number 2 
+using 3 bits.  We could send 2 in binary using 2 bits `10`, but we need 3 bits 
+to comply with the protocol.
+
+Finally we include 12 bits corresponding with the 1 bit between request & 
+response, the 1 null bit that is zero, and the 10 bits returning the result.  The
+ADC does not care about these bits so we just send 12 `0` bits.
+
+Now we'll pattern match the response to get the returned value. 
+
+    <<_::size(14), value::size(10)>> =
```

Comment:
I'm pretty sure this should be `<<_::size(12, value::size(12)>>` since the ADC is providing 12 bits and the code is now getting all of the bits. The previous code would ignore the low 2 bits - hence it became a 10 bit ADC.